js_modules/dagit/packages/core/src/nav/useRepositoryLocationReload.tsx

```diff
@@ -79,37 +106,58 @@ export const useRepositoryLocationReload = (location: string) => {
         const workspace = data.workspaceOrError;
 
         if (workspace.__typename === 'PythonError') {
-          dispatch({type: 'error', error: workspace});
+          dispatch({type: 'error', error: workspace, errorLocationId: null});
           stopPolling();
           return;
         }
+        if (state.pollLocationIds === null) {
+          stopPolling();
+          return;
+        }
+
+        const matches = state.pollLocationIds
+          .map((location) => workspace.locationEntries.find((l) => l.id === location)!)
+          .filter(Boolean);
+        const missing = state.pollLocationIds.find((id) => !matches.some((m) => m.id === id));
 
-        const match = workspace.locationEntries.find((l) => l.id === location);
-        if (!match) {
+        if (missing) {
           dispatch({
             type: 'error',
-            error: {message: `Location ${location} not found in workspace.`},
+            error: {message: `Location ${missing} not found in workspace.`},
+            errorLocationId: missing,
           });
           stopPolling();
           return;
         }
 
         // If we're still loading, there's nothing to do yet. Continue polling unless
         // we have hit our timeout threshold.
-        if (match.loadStatus === RepositoryLocationLoadStatus.LOADING) {
+        if (matches.some((l) => l.loadStatus === RepositoryLocationLoadStatus.LOADING)) {
           if (Date.now() - Number(state.pollStartTime) > THREE_MINUTES) {
+            const message = `Timed out waiting for the location${
+              matches.length > 1 ? 's' : ''
+            } to reload.`;
```

Comment:
I think I'm a little bit allergic to handling plurals this way. 😆 

I know it doesn't matter until we i18n-ize the app and this wouldn't be completely correct in an i18n case anyway, but...mind making this `{conditional ? 'location' : 'locations'}` instead?